sycl/source/detail/plugin.hpp

```diff
@@ -184,11 +184,67 @@ class plugin {
   void *getLibraryHandle() { return MLibraryHandle; }
   int unload() { return RT::unloadPlugin(MLibraryHandle); }
 
+  // return the index of PiPlatforms.
+  // If not found, add it and return its index.
+  int getPlatformId(RT::PiPlatform Platform) {
+    if (PiPlatforms) {
```

Comment:
Why this function can be called with `PiPlatforms == nullptr`? Can we replace it with `assert(PiPlatforms != nullptr)`?

Reply:
I changed the code to check if nullptr and allocated a memory.
PiPlatforms are nullptr when RT::initialize() is not called yet.